### PR TITLE
bugfix: set correct default values to 3d pooling padding and stride

### PR DESCRIPTION
**Pull request by ckomaki**
Currently I'm experiencing this weird behavior below.

```
>>> import mxnet as mx
>>> import numpy as np
>>>
>>> def test_pooling(pooling):
...     pooling_ex = pooling.simple_bind(mx.gpu())
...     pooling_ex.arg_dict['a'][:] = np.arange(8).reshape((1, 1, 2, 2, 2))
...     return pooling_ex.forward()[0].asnumpy()
...
>>> a = mx.sym.Variable('a', shape=(1, 1, 2, 2, 2))
>>>
>>> pad = mx.sym.Pooling(data=a, kernel=(2, 2, 2), stride=(1, 1, 1), pad=(0, 0, 0), pool_type='max')
>>> print test_pooling(pad)
[[[[[ 7.]]]]]
>>> 
>>> no_pad = mx.sym.Pooling(data=a, kernel=(2, 2, 2), stride=(1, 1, 1), pool_type='max')
>>> print test_pooling(no_pad)
[[[[[ -3.40282347e+38]]]]]
```

### CODE REVIEW

File: src/operator/pooling-inl.h [hunk 1]
```diff
@@ -57,13 +57,13 @@ struct PoolingParam : public dmlc::Parameter<PoolingParam> {
               "kValid is default setting of Mxnet and rounds down the output pooling size."
               "kFull is compatible with Caffe and rounds up the output pooling size.");
 
-    int stride_shape[] = {1, 1};
-    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + 2))
+    int stride_shape[] = {1, 1, 1};
+    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + kernel.ndim()))
```
Comment (piiswrong) [inline src/operator/pooling-inl.h]: Set default to TShape() here and set it to 1, 1 or 1, 1, 1 depending on kernel dim in infershape

Comment (ckomaki) [inline src/operator/pooling-inl.h]: Can I set shapes in PoolingProp::InferShape? It is a const function.

I can set them in PoolingProp::Init as convolution-inl.h does. 
https://github.com/dmlc/mxnet/blob/898c4e71f1d21aae2cae5a86eb9ed15c1c01ada1/src/operator/convolution-inl.h#L354

Comment (piiswrong) [inline src/operator/pooling-inl.h]: yes that works too

Comment (ckomaki) [inline src/operator/pooling-inl.h]: Moved and committed.

File: src/operator/pooling-inl.h [hunk 2]
```diff
@@ -183,7 +181,16 @@ Operator* CreateOp(PoolingParam param, int dtype);
...
  public:
   void Init(const std::vector<std::pair<std::string, std::string> >& kwargs) override {
+    using namespace mshadow;
     param_.Init(kwargs);
+    if (param_.kernel.ndim() == 2) {
+      if (param_.stride.ndim() == 0) param_.stride = Shape2(1, 1);
+      if (param_.pad.ndim() == 0) param_.pad = Shape2(0, 0);
+    } else {
+      CHECK_EQ(param_.kernel.ndim(), 3) << param_.kernel.ndim() << "D pooling not supported";
+      if (param_.stride.ndim() == 0) param_.stride = Shape3(1, 1, 1);
```
Comment (piiswrong) [inline src/operator/pooling-inl.h]: if param_.stride.ndim() != 0 is there check for correct ndim in infershape? If not we should check here.

Comment (ckomaki) [inline src/operator/pooling-inl.h]: Added checks at the bottom.

**Comment by piiswrong** [comment]
Please fix lint: src/operator/pooling-inl.h:194:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
src/operator/pooling-inl.h:195:  Lines should be <= 100 characters long  [whitespace/line_length] [2]

**Comment by piiswrong** [comment]
merged. Thanks!

**Pull request merged**